src/scripts/main.js

```diff
@@ -1,3 +1,59 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table.field');
+const buttons = document.querySelectorAll('button.button');
+
+buttons.forEach((button) => {
+  button.addEventListener('click', () => {
+    const tbody = table.querySelector('tbody') || table;
+
+    if (button.classList.contains('append-row')) {
+      if (table.rows.length >= 10) {
```

Comment:
The condition `table.rows.length >= 10` is used to prevent adding more than 10 rows. Ensure that this limit aligns with your requirements. If the maximum number of rows should be different, adjust this value accordingly.